### ISSUE TITLE
Supervised and Unsupervised split training

### ISSUE THREAD

**Issue by debasmitdas** [zhenyuw16/noisyboundaries#7]
Thank you for this great work. I wanted to know whether the losses used through NTM and BPM are applied to both labelled and unlabelled data ? Or is it that it is only applied to unlabelled data and for the labelled data it is the regular loss function.
